### PR TITLE
London 9 - Andriana Saffo - JS2 - Week 4

### PR DESCRIPTION
**Pull request by AndrianaOS**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Andriana Saffo
- Your City: London
- Your Slack Name: Andriana

# Homework Details

- Module: JS 2
- Week: 4

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 1]
```diff
@@ -1,4 +1,33 @@
...
+
+  let minutes = Math.floor(countdownTime / 60);
+
+  let seconds = countdownTime - minutes * 60;
+
+  timeRemaining.innerText = `Time Remaining: 0${minutes}:${seconds}`;
+
+  let countdown = setInterval(() => {
+    seconds -= 1;
+    if (seconds == 0) {
```
Comment (VitalinaKuzmenko) [inline 1-mandatory/1-alarmclock/alarmclock.js:15]:
```suggestion
    if (seconds === 0 && minutes === 0) {
```

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 2]
```diff
@@ -1,4 +1,33 @@
...
+    }
+    if (seconds < 10) {
+      seconds = "0" + seconds;
+      // minutes = "0" + minutes;
+    } else {
+      seconds = seconds;
+      minutes = minutes;
+    }
+
+    timeRemaining.innerText = `Time Remaining: 0${minutes}:${seconds}`;
```
Comment (VitalinaKuzmenko) [inline 1-mandatory/1-alarmclock/alarmclock.js:28]:
```suggestion
    timeRemaining.innerText = `Time Remaining: 0${minutes}:${seconds}`;
    countdownTime--;
```

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 3]
```diff
@@ -1,4 +1,33 @@
...
+  let countdownTime = timeInput.value;
+
+  let minutes = Math.floor(countdownTime / 60);
+
+  let seconds = countdownTime - minutes * 60;
+
+  timeRemaining.innerText = `Time Remaining: 0${minutes}:${seconds}`;
+
+  let countdown = setInterval(() => {
+    seconds -= 1;
```
Comment (VitalinaKuzmenko) [inline 1-mandatory/1-alarmclock/alarmclock.js:14]:
```suggestion
  let minutes = Math.floor(countdownTime / 60);
  let seconds = countdownTime - minutes * 60;
```

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 4]
```diff
@@ -1,4 +1,33 @@
-function setAlarm() {}
+function setAlarm() {
+  const timeInput = document.querySelector("#alarmSet");
+
+  let timeRemaining = document.querySelector("#timeRemaining");
+  let countdownTime = timeInput.value;
+
+  let minutes = Math.floor(countdownTime / 60);
```
Comment (VitalinaKuzmenko) [inline 1-mandatory/1-alarmclock/alarmclock.js:7]:
```suggestion

```

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 5]
```diff
@@ -1,4 +1,33 @@
...
+
+  let seconds = countdownTime - minutes * 60;
```
Comment (VitalinaKuzmenko) [inline 1-mandatory/1-alarmclock/alarmclock.js:9]:
```suggestion

```

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 6]
```diff
@@ -1,4 +1,33 @@
...
+
+  timeRemaining.innerText = `Time Remaining: 0${minutes}:${seconds}`;
```
Comment (VitalinaKuzmenko) [inline 1-mandatory/1-alarmclock/alarmclock.js:11]:
```suggestion

```

**Review by VitalinaKuzmenko — CHANGES_REQUESTED** [review]
Hi, Andriana! I have a time to check your alarmclock exercise. First of all, I want to say great job! I see that you have great knowledge at Javascript. Becuase this exercise was really very-very hard! Well done!
I found out that your alarm is not working properly. That's why I decided to suggest some changes to make it work.
My logic is:
1) You checking what is the value inside input. Creating "countdownTime" variable.
2)  Then inside countdown variable you using setInterval() method
3) inside this method you're making from input minutes and secons.
4) Then if seconds are equal to 0 AND minutes are equal to 0 - you stop the countdown. (You were checking only seconds. But in case when you have 02:00 (2 minutes and 0 seconds) your alarm will be stop but it shouldn't).
5) if seconds are less 10 - you adding 0 to seconds. (That was right!(
6) Then you changing innerText of our text on webpage.
7) At the end, we're reducing our countdownTime value  by 1. (In this case. If the user will provide number 65. Inside setinterval method, we have 1 minute and 5 seconds, changing innerText and then reducing input value. So we have have 64. Inside setInterval method we making minutes and seconds and we already have 1 minute and 4 seconds.)
You did really great job! Try to change my suggestions and check if it's working now.

I also attached an image of already changed functions. (maybe it will be easier for you to understand).
Let me know If you need anything else or if you don't understand anything!

andriana-alarmClock

File: 1-mandatory/1-alarmclock/alarmclock.js (hunk 6 above)
Comment (AndrianaOS) [reply 1-mandatory/1-alarmclock/alarmclock.js:11]: I will implement your suggestions. Thank you Vitalina 😊

**Pull request closed without merging**